### PR TITLE
Added registerPostProcessingTask() to the DSL.

### PR DESCRIPTION
**Pull request by autonomousapps**

### CODE REVIEW

File: src/main/kotlin/com/autonomousapps/AbstractPostProcessingTask.kt [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+import com.autonomousapps.internal.utils.fromJson
+import org.gradle.api.DefaultTask
+import org.gradle.api.file.RegularFileProperty
+import org.gradle.api.tasks.InputFile
+import org.gradle.api.tasks.PathSensitive
+import org.gradle.api.tasks.PathSensitivity
+
+abstract class AbstractPostProcessingTask : DefaultTask() {
+
+  @get:PathSensitive(PathSensitivity.NONE)
```
Comment (ZacSweers) [inline src/main/kotlin/com/autonomousapps/AbstractPostProcessingTask.kt:15]: Why none?

**Comment by autonomousapps** [comment]
Only the content of the file matters. The path to it and name of it are
irrelevant.

File: src/main/kotlin/com/autonomousapps/DependencyAnalysisPlugin.kt [hunk 2]
```diff
@@ -677,6 +677,20 @@ class DependencyAnalysisPlugin : Plugin<Project> {
         outputPretty.set(paths.aggregateAdvicePrettyPath)
       }
 
+    // Is there a post-processing task? If so, run it
+    afterEvaluate {
+      val postProcessingTask = if (this == rootProject) {
```
Comment (sboishtyan) [inline src/main/kotlin/com/autonomousapps/DependencyAnalysisPlugin.kt:682]: Could we replace `afterEvaluate` if we used Gradle property or provider for postProcessingTask?

Comment (autonomousapps) [reply src/main/kotlin/com/autonomousapps/DependencyAnalysisPlugin.kt:682]: I would have liked to do that, but I don't know if a `TaskProvider<Something>` can be held in a property.

There's also the issue that I do this wiring at configuration time and not task execution time, so I have to "wait" (via `afterEvaluate`) for the value to be set on the extension before I can query it.

**Pull request merged**